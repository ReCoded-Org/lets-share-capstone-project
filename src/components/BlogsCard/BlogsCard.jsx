import React from "react";
import Image from "next/image";
import Button from "../button";
import Link from "next/link";
import { useTranslation } from "next-i18next";

export default function BlogsCard({ item }) {
    const { t } = useTranslation("common");
    const index = item.id;
    const evenindex = index % 2 === 0 ? true : false;

    return (
        <div>
            <div
                className={
                    evenindex
                        ? "flex flex-col items-center px-5 font-primary lg:mt-5 lg:h-[60vh] lg:flex-row lg:items-center xl:mx-20 "
                        : "flex flex-col items-center px-5 font-primary lg:mt-5 lg:h-[60vh] lg:flex-row-reverse lg:items-center xl:mx-20 "
                }
            >
                <div className='relative w-full p-10 px-10'>
                    <Image
                        src={item.image}
                        className='card-img '
                        alt={item.imageAlt}
                        width={1200}
                        height={800}
                    />
                </div>

                <div className=' flex flex-col items-start justify-center'>
                    <div className=' px-10'>
                        <h2 className='pb-2 font-head text-3xl font-bold tracking-wider text-fontColor'>
                            {t(`blogs.headerBlg${item.id}`)}
                        </h2>
                        <p className='text-lg text-fontColor'>
                            {t(`blogs.contentBlg${item.id}`).slice(0, 250) +
                                "..."}
                        </p>

                        <div className='flex h-7 w-10 flex-1'>
                            <Link href={`/blogs/${item.title.toString()}`}>
                                <a>
                                    <Button fullfilled='Read more' />
                                </a>
                            </Link>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
