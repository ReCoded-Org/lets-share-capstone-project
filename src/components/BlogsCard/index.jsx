import BlogsCard from "./BlogsCard";

export default BlogsCard;
