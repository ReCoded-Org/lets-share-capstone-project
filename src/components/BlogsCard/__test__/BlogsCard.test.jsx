import renderer from "react-test-renderer";
import BlogsCard from "../BlogsCard";
import * as nextRouter from "next/router";
const item = {
    id: 2,
    title: "My Blog",
    description:
        " Lorem ipsum, dolor sit amet consectetur adipisicing elit. Illum necessitatibus numquam qui ea debitis natus amet ab nobis dignissimos consequatur possimus obcaecati corporis doloremque velit, temporibus praesentium unde sed? Quae?",
    image: "/Photo.png",
    date: "01/01/2015",
};
it("renders correctly", () => {
    const tree = renderer.create(<BlogsCard item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
});
function tryRoute(packacge) {
    packacge.useRouter = jest.fn();
    packacge.useRouter.mockImplementation(() => ({ route: "/" }));
}
tryRoute(nextRouter);
