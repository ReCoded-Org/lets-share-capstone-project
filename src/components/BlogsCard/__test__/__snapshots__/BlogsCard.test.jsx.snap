// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="mb-3 flex  flex-row justify-between p-10"
  >
    <div
      className=" p-10 px-10"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271000%27%20height=%27800%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="Card image"
          className="card-img"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Card image"
            className="card-img"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2FPhoto.png&w=2048&q=75"
            srcSet="/_next/image?url=%2FPhoto.png&w=1080&q=75 1x, /_next/image?url=%2FPhoto.png&w=2048&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
    <div
      className=" flex flex-col items-start justify-center"
    >
      <div
        className="r-md-8 px-10"
      >
        <h2
          className="pb-2 text-xl font-bold text-fontColor"
        >
          My Blog
        </h2>
        <p
          className="text-lg text-fontColor"
        >
           Lorem ipsum, dolor sit amet consectetur adipisicing elit. Illum necessitatibus numquam qui ea debitis natus amet ab nobis dignissimos consequatur possimus obcaecati corporis doloremque velit, temporibus praesentium unde sed? Quae?
        </p>
        <div
          className="flex h-7 w-10 flex-1"
        >
          <div
            className="flex justify-center"
          >
            <button
              className="my-5 h-9 w-[120px] rounded-full bg-primary text-xl font-semibold text-[white] shadow-md"
            >
              Read more
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
